src/libcollections/vec.rs

```diff
@@ -1380,7 +1380,17 @@ impl<T> FromIterator<T> for Vec<T> {
     fn from_iter<I:Iterator<Item=T>>(iterator: I) -> Vec<T> {
         let (lower, _) = iterator.size_hint();
         let mut vector = Vec::with_capacity(lower);
-        for element in iterator {
+
+        let mut i = iterator.fuse();
+        for element in i.by_ref().take(vector.capacity()) {
+            let len = vector.len();
```

Comment:
Since `len` is just a field, this code is basically just read/writing to the Vec's len field as a local variable, without the need for an actual local variable. (and setting the len in the loop is important for guarding against panics)